File: .pub-cache/hosted/pub.dartlang.org/socket_io_client-2.0.2/test/io_client.dart
// Copyright (C) 2019 Potix Corporation. All Rights Reserved
// History: 2019-01-21 11:56
// Author: jumperchen<jumperchen@potix.com>
import 'dart:async';
import 'package:socket_io_client/socket_io_client.dart' as io;

void main() {
  var socket = io.io(
      'http://localhost:3000',
      io.OptionBuilder()
          .setTransports(['websocket'])
          // .disableAutoConnect()
          .build());

  // socket.connect();

  socket.onConnect((_) {
    socket.emit('toServer', 'init');

    var count = 0;
    Timer.periodic(const Duration(seconds: 1), (Timer countDownTimer) {
      socket.emit('toServer', count++);
    });
  });

  socket.on('event', (data) => print(data));
  socket.on('disconnect', (_) => print('disconnect'));
  socket.on('fromServer', (_) => print(_));
}

File: .pub-cache/hosted/pub.dartlang.org/socket_io_client-2.0.2/test_nodejs/io_client.js
/**
 * io_client.js
 *
 * Purpose:
 *
 * Description:
 *
 * History:
 *    2020/5/29, Created by jumperchen
 *
 * Copyright (C) 2020 Potix Corporation. All Rights Reserved.
 */
'use strict';
const io = require('socket.io-client');

var socket = io('http://localhost:3000', {
    transports: ['websocket'],
    autoConnect: false
});
socket.on('connect', function(){
    console.log('connect');
    socket.emit('msg', 'hi')
});
socket.connect();
socket.on('connect_error', function(){console.log('connect_error')});
socket.on('event', function(data){});
socket.on('disconnect', function(){console.log('disconnect')});
